hw/ip/hmac/dv/cov/hmac_cov_unr_excl.el

```diff
@@ -0,0 +1,141 @@
+CHECKSUM: "2024123546 95776463"
+INSTANCE: tb.dut.u_tlul_adapter.u_sramreqfifo
+ANNOTATION: "VC_COV_UNR"
+Condition 1 "2331864568" "(gen_normal_fifo.full ? (1'(Depth)) : ((gen_normal_fifo.wptr_msb == gen_normal_fifo.rptr_msb) ? ((1'(gen_normal_fifo.wptr_value) - 1'(gen_normal_fifo.rptr_value))) : (((1'(Depth) - 1'(gen_normal_fifo.rptr_value)) + 1'(gen_normal_fifo.wptr_value))))) 1 -1" (2 "1")
+ANNOTATION: "VC_COV_UNR"
+Condition 2 "1926118060" "((gen_normal_fifo.wptr_msb == gen_normal_fifo.rptr_msb) ? ((1'(gen_normal_fifo.wptr_value) - 1'(gen_normal_fifo.rptr_value))) : (((1'(Depth) - 1'(gen_normal_fifo.rptr_value)) + 1'(gen_normal_fifo.wptr_value)))) 1 -1" (1 "0")
+ANNOTATION: "VC_COV_UNR"
+Condition 3 "2329396864" "(gen_normal_fifo.empty ? 'b0 : gen_normal_fifo.rdata_int) 1 -1" (1 "0")
+CHECKSUM: "3042386013 3871380215"
+INSTANCE: tb.dut.u_tlul_adapter
+ANNOTATION: "VC_COV_UNR"
+Condition 3 "800561441" "((vld_rd_rsp && reqfifo_rdata.error) ? error_blanking_data : (vld_rd_rsp ? rspfifo_rdata.data : '0)) 1 -1" (2 "1")
+ANNOTATION: "VC_COV_UNR"
+Condition 4 "2745829604" "(vld_rd_rsp && reqfifo_rdata.error) 1 -1" (2 "10")
+ANNOTATION: "VC_COV_UNR"
+Condition 4 "2745829604" "(vld_rd_rsp && reqfifo_rdata.error) 1 -1" (3 "11")
+ANNOTATION: "VC_COV_UNR"
+Condition 5 "1119953630" "(vld_rd_rsp ? rspfifo_rdata.data : '0) 1 -1" (2 "1")
+ANNOTATION: "VC_COV_UNR"
+Condition 6 "561780173" "((vld_rd_rsp && reqfifo_rdata.error) ? error_blanking_integ : (vld_rd_rsp ? rspfifo_rdata.data_intg : prim_secded_pkg::SecdedInv3932ZeroEcc)) 1 -1" (2 "1")
+ANNOTATION: "VC_COV_UNR"
+Condition 7 "3130851184" "(vld_rd_rsp && reqfifo_rdata.error) 1 -1" (2 "10")
+ANNOTATION: "VC_COV_UNR"
+Condition 7 "3130851184" "(vld_rd_rsp && reqfifo_rdata.error) 1 -1" (3 "11")
+ANNOTATION: "VC_COV_UNR"
+Condition 8 "2942914969" "(vld_rd_rsp ? rspfifo_rdata.data_intg : prim_secded_pkg::SecdedInv3932ZeroEcc) 1 -1" (2 "1")
+ANNOTATION: "VC_COV_UNR"
+Condition 13 "1798941048" "(d_valid && d_error) 1 -1" (1 "01")
+ANNOTATION: "VC_COV_UNR"
+Condition 16 "4228513916" "((((reqfifo_rdata.op == OpRead) & (~reqfifo_rdata.error))) ? reqfifo_rready : 1'b0) 1 -1" (2 "1")
+CHECKSUM: "1413060523 1581044118"
+INSTANCE: tb.dut.u_sha2
+Fsm sha_st_q "2028159779"
+ANNOTATION: "VC_COV_UNR"
+Transition ShaUpdateDigest->ShaCompress "2->1"
+CHECKSUM: "2024123546 2243903072"
+INSTANCE: tb.dut.u_tlul_adapter.u_reqfifo
+ANNOTATION: "VC_COV_UNR"
+Branch 0 "1862733684" "gen_normal_fifo.full" (2) "gen_normal_fifo.full 0,0"
+ANNOTATION: "VC_COV_UNR"
+Branch 3 "2807735941" "(!rst_ni)" (1) "(!rst_ni) 0,1,-,-"
+ANNOTATION: "VC_COV_UNR"
+Branch 3 "2807735941" "(!rst_ni)" (3) "(!rst_ni) 0,0,1,0"
+ANNOTATION: "VC_COV_UNR"
+Branch 4 "1287846560" "(!rst_ni)" (1) "(!rst_ni) 0,1,-,-"
+ANNOTATION: "VC_COV_UNR"
+Branch 4 "1287846560" "(!rst_ni)" (3) "(!rst_ni) 0,0,1,0"
```

Comment:
These excludes the `always_ff` block due to the format.
I think the tool is looking for 
```
  always_ff (...) begin
    if (!rst_ni) begin
       .......
    end else begin
      ........   --> it says this part is missing
   end
end
```
Reference code: https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_fifo_sync.sv#L97

Reply:
Sorry. I don't understand this. I don't know why the tool can't reach a certain part of the block.

Reply:
No worries, I did not explain this well.
The coverage tool (without) expects this:
```
    always_ff @(posedge clk_i or negedge rst_ni) begin
      if (!rst_ni) begin
        fifo_wptr <= {(PTR_WIDTH){1'b0}};
      end else if (clr_i) begin
        fifo_wptr <= {(PTR_WIDTH){1'b0}};
      end else if (fifo_incr_wptr) begin
        if (fifo_wptr[PTR_WIDTH-2:0] == (PTR_WIDTH-1)'(Depth-1)) begin
          fifo_wptr <= {~fifo_wptr[PTR_WIDTH-1],{(PTR_WIDTH-1){1'b0}}};
        end else begin
          fifo_wptr <= fifo_wptr + {{(PTR_WIDTH-1){1'b0}},1'b1};
        end
      end else begin
         XXX something default  ---> The coverage tool report this as missing branch :)
      end
    end
```
All of them are saying the "else" part is missing, and report them as missing branch. And UNR comes and excluded these missing branch, because the final `else` branch is unreachable.

Reply:
I think the coverage tool won't say there is any missing branch if we code it like this:
```
    always_ff @(posedge clk_i or negedge rst_ni) begin
      if (!rst_ni) begin
        fifo_wptr <= {(PTR_WIDTH){1'b0}};
      end else begin
        if (clr_i) begin
          fifo_wptr <= {(PTR_WIDTH){1'b0}};
          end else if (fifo_incr_wptr) begin
          if (fifo_wptr[PTR_WIDTH-2:0] == (PTR_WIDTH-1)'(Depth-1)) begin
            fifo_wptr <= {~fifo_wptr[PTR_WIDTH-1],{(PTR_WIDTH-1){1'b0}}};
          end else begin
            fifo_wptr <= fifo_wptr + {{(PTR_WIDTH-1){1'b0}},1'b1};
          end
      end 
    end
```

Reply:
hum. as this logic is in `always_ff`, missing `else` shouldn't be a problem. I am ok to waive this.